iris-mpc-cpu/src/hawkers/plaintext_store.rs

```diff
@@ -1,129 +1,95 @@
+
+        (code_distance as u16, combined_mask_len as u16)
     }
-}
 
-impl FormattedIris {
-    #[cfg(test)]
-    pub fn compute_real_distance(&self, other: &FormattedIris) -> f64 {
-        let hd = self.dot_on_code(other);
-        let mask_ones = (self.mask & other.mask).count_ones();
-        ((mask_ones as f64) - (hd as f64)) / (2. * mask_ones as f64)
+    /// Return the fractional Hamming distance with another PlaintextIris,
+    /// represented as the i16 dot product of associated masked-bit vectors
+    /// and the u16 size of the common unmasked region
+    pub fn dot_distance_fraction(&self, other: &Self) -> (i16, u16) {
+        let (code_distance, combined_mask_len) = self.distance_fraction(other);
+        let dot_product = combined_mask_len.wrapping_sub(2 * code_distance) as i16;
+
+        (dot_product, combined_mask_len)
     }
 }
 
 #[derive(Clone, Default, Debug)]
 pub struct PlaintextPoint {
     /// Whatever encoding of a vector.
-    data:          FormattedIris,
+    data:          PlaintextIris,
     /// Distinguish between queries that are pending, and those that were
     /// ultimately accepted into the vector store.
     is_persistent: bool,
 }
 
-impl FormattedIris {
-    pub fn dot_on_code(&self, other: &Self) -> i32 {
-        self.data
-            .iter()
-            .zip(other.data.iter())
-            .fold(0_i32, |sum, (i, j)| sum + (*i as i32) * (*j as i32))
-    }
-    pub fn compute_distance(&self, other: &Self) -> (i16, usize) {
-        let combined_mask = self.mask & other.mask;
-        let dot = self.dot_on_code(other) as i16;
-        (dot, combined_mask.count_ones())
-    }
-}
-
-impl PlaintextPoint {
-    fn compute_distance(&self, other: &PlaintextPoint) -> (i16, usize) {
-        self.data.compute_distance(&other.data)
-    }
-
-    fn is_close(&self, other: &PlaintextPoint) -> bool {
-        let hd = self.data.dot_on_code(&other.data);
-        let mask_ones = (self.data.mask & other.data.mask).count_ones();
-        let threshold = (mask_ones as f64) * (1. - 2. * MATCH_THRESHOLD_RATIO);
-        (threshold as i32 - hd) < 0
-    }
-}
-
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Serialize, Deserialize)]
-pub struct PointId(pub usize);
-
-impl PointId {
-    pub fn val(&self) -> usize {
-        self.0
-    }
-}
+pub type PointId = u32;
 
 impl PlaintextStore {
     pub fn prepare_query(&mut self, raw_query: IrisCode) -> <Self as VectorStore>::QueryRef {
         self.points.push(PlaintextPoint {
-            data:          FormattedIris::from(raw_query),
+            data:          PlaintextIris(raw_query),
             is_persistent: false,
         });
 
         let point_id = self.points.len() - 1;
-        PointId(point_id)
+        point_id as PointId
     }
 
+    /// Compare two distances between pairs of iris codes
+    ///
+    /// Returns terms obtained by cross-multiplying numerators with opposite
+    /// denominators.
     pub fn distance_computation(
```

Comment:
It looks that this function isn't used anywhere. Plus, it is inconsistent with the new `DistanceRef` of `PlaintextStore`. You can safely delete it.